addons/escoria-core/game/core-scripts/behaviors/esc_movable.gd

```diff
@@ -379,27 +383,111 @@ func is_angle_in_interval(angle: float, direction_angle: ESCDirectionAngle) -> b
 #
 # - deg int angle to set the character
 # - immediate
-#	If true, direction is switched immediately. Else, successive animations are
-#	used so that the character turns to target angle.
+#	If true, direction is switched immediately. Else, successive 
+#	animations are used so that the character turns to target angle.
 func set_angle(deg: int, immediate = true) -> void:
 	if deg < 0 or deg > 360:
 		escoria.logger.report_errors(
 			"movable.gd:set_angle()",
 			["Invalid degree to turn to " + str(deg)]
 		)
 	moved = true
-	last_deg = deg
-	last_dir = _get_dir_deg(deg, parent.animations)
-
-	# The character may have a state animation from before, which would be
-	# resumed, so we immediately force the correct idle animation
-	if parent.animation_sprite.animation != \
-			parent.animations.idles[last_dir].animation:
-		parent.animation_sprite.play(parent.animations.idles[last_dir].animation)
-	pose_scale = -1 if parent.animations.idles[last_dir].mirrored else 1
+	
+	if immediate:
+		last_deg = deg
+		last_dir = _get_dir_deg(deg, parent.animations)
+
+		# The character may have a state animation from before, which would be
+		# resumed, so we immediately force the correct idle animation
+		if parent.animation_sprite.animation != \
+				parent.animations.idles[last_dir].animation:
+			parent.animation_sprite.play(
+				parent.animations.idles[last_dir].animation
+			)
+		pose_scale = -1 if parent.animations.idles[last_dir].mirrored else 1
+	else:
+		var current_dir = last_dir
+		last_deg = deg
+		var target_dir = _get_dir_deg(deg, parent.animations)
+		
+		if target_dir - current_dir > 1:
+			var way_to_turn = get_shortest_way_to_dir(current_dir, target_dir)
+		
+			var dir = current_dir
+			while dir != target_dir:
+				dir += way_to_turn
+				if dir >= parent.animations.dir_angles.size():
+					dir = 0
+				if dir < 0:
+					dir = parent.animations.dir_angles.size() - 1
+				
+#				if parent.animation_sprite.animation != \
+#						parent.animations.idles[last_dir].animation:
+				parent.animation_sprite.play(
+					parent.animations.idles[dir].animation
+				)
+				yield(parent.animation_sprite, "animation_finished")
+				pose_scale = -1 if parent.animations.idles[dir].mirrored else 1
+			
+				
+			
+		else:
```

Comment:
This is duplicated. Better change the condition in line 396 so that immediate changes and 1-step direction changes use the same code.